test/shared/src/main/scala/zio/test/DefaultTestReporter.scala

```diff
@@ -144,9 +146,41 @@ object DefaultTestReporter {
   private def renderCause(cause: Cause[Any], offset: Int): String =
     cause match {
       case Cause.Die(TestTimeoutException(message)) => message
-      case _                                        => cause.prettyPrint.split("\n").map(withOffset(offset + tabSize)).mkString("\n")
+      case Cause.Traced(Cause.Die(ex: MockException), _) =>
+        renderMockException(ex).split("\n").map(withOffset(offset + tabSize)).mkString("\n")
+      case _ => cause.prettyPrint.split("\n").map(withOffset(offset + tabSize)).mkString("\n")
     }
 
+  private def renderMockException(exception: MockException): String =
+    exception match {
+      case InvalidArgumentsException(method, assertResult) =>
+        Seq(
+          red(s"Method $method called with invalid arguments:"),
+          renderAssertResult(assertResult)
+        ).mkString("\n")
+
+      case InvalidMethodException(method, expected, assertion) =>
+        List(
+          red(s"Invalid call to $method, expected $expected"),
+          withOffset(tabSize)(s"with arguments ${cyan(renderAssertion(assertion))}")
+        ).mkString("\n")
+
+      case UnexpectedCallException(method, args) =>
+        List(
+          red(s"Unexpected call to $method"),
+          withOffset(tabSize)(s"with arguments ${blue(args.toString)}")
+        ).mkString("\n")
+    }
+
+  private def renderAssertResult(assertResult: AssertResult): String =
```

Comment:
Are you confident that there is never going to be an `AssertResult` with multiple failures (e.g. `p && q` where both `p` and `q` are false)? If there is one it is not going to match your pattern and you are not going to get any reporting. You may want to do something like:

```scala
private def renderAssertResult(assertResult: AssertResult): String =
  assertResult.failures.fold("")(
    _.fold(renderFragment(value, 0))(
      "(" + _ + " && " + _ + ")",
      "(" + _ + " || " + _ + ")",
      "~" + _
    )
  )
```

Reply:
That got me thinking, and I replaced `assertion.test(input)` with `assert(input, assertion)`, this way the exception holds `TestResult` instead of `AssertResult` and we can reuse the same rendering mechanism as in regular assertions.

Reply:
Nice!

Reply:
Nice!